pkg/operator/ceph/cluster/cluster.go

```diff
@@ -595,28 +600,86 @@ func (c *cluster) reportTelemetry() {
 	telemetry.ReportKeyValue(c.context, c.ClusterInfo, telemetry.ExternalModeEnabledKey, strconv.FormatBool(c.Spec.External.Enable))
 }
 
+func (c *cluster) addSettingsToMonDB(clientName string, settings map[string]string) error {
+	root := path.Join(c.context.ConfigDir, c.Namespace)
+	if err := os.MkdirAll(root, 0744); err != nil {
+		return errors.Wrapf(err, "failed to create assimilateConfPath for %s.", clientName)
+	}
+
+	assimilateConfPath := path.Join(root, fmt.Sprintf("%s.txt", clientName))
```

Comment:
This method may potentially run in the future from multiple goroutines, so we need to have a unique input file. How about using `ioutil.TempFile()` to generate the filename to avoid that possible conflict?

Reply:
make sense. Done!